appengine/py/appengine_deploy.sh.template

```diff
@@ -0,0 +1,55 @@
+#!/bin/bash
+# Copyright 2017 The Bazel Authors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+case "$0" in
+/*) self="$0" ;;
+*)  self="$PWD/$0";;
+esac
+if [[ -e "$self.runfiles/%{workspace_name}" ]]; then
+  RUNFILES="$self.runfiles/%{workspace_name}"
+  cd $RUNFILES
+fi
+
+ROOT=$PWD
+tmp_dir=$(mktemp -d ${{TMPDIR:-/tmp}}/war.XXXXXXXX)
+cp -R $ROOT $tmp_dir
+trap "{{ cd ${{root_path}}; rm -rf $tmp_dir; }}" EXIT
+rm -Rf $tmp_dir/%{workspace_name}/external/com_google_appengine_py
+if [ -n "${{1-}}" ]; then
+  has_app_yaml=$(echo ${{@:2}} | grep -E "^([^ ]+ +)*app.yaml")
+  other_mod_cfgs=$(echo ${{@:2}} | xargs printf "%s\n" | grep -v "^app\.yaml$" | xargs echo)
+  ret_code=0
+  # Secondary modules need to be uploaded first because if any of them are
+  # referenced in dispatch.yaml, App Engine must have a version of that module
+  # prior to the app.yaml upload.
+  if [ -n "$other_mod_cfgs" ]; then
+    (cd $tmp_dir/%{workspace_name} && $ROOT/%{appcfg} -A "$1" update $other_mod_cfgs)
+    ret_code=$?
+  fi
+  if [ $ret_code -eq 0 ] && [ -n "$has_app_yaml" ] || [ -z "$other_mod_cfgs" ]; then
+    $ROOT/%{appcfg} -A "$1" update $tmp_dir/%{workspace_name}
+    ret_code=$?
+  fi
+
+else
+  echo "\033[1;31mERROR:\033[0m Application ID must be provided as first argument
+  USAGE: bazel run path/to/my/gae_binary_target.deploy -- my-project-id [module.yaml files ...]"
+  ret_code=-1
+fi
+
+rm -Rf $tmp_dir
+trap - EXIT
```

Comment:
Why not just let the exit handler delete tmp_dir?

Reply:
I didn't write this, so I don't know.

Reply:
The trap - EXIT deletes the trap above so the rm defined there doesn't run on exit. This was also copied from the java counterpart, but I don't know why it would replicate the trap. I've removed it in my change and we can discuss it there.